pipeline_dp/accumulator.py

```diff
@@ -108,6 +124,33 @@ def compute_metrics(self):
         ]
 
 
+class AccumulatorFactory:
+    """Factory for producing the appropriate Accumulator depending on the
+    AggregateParams and BudgetAccountant."""
+
+    def __init__(self, params: pipeline_dp.AggregateParams,
+                 budget_accountant: pipeline_dp.BudgetAccountant):
+        self._params = params
+        self._budget_accountant = budget_accountant
+
+    def initialize(self):
+        self._accumulator_params = create_accumulator_params(
+            self._params, self._budget_accountant)
+
+    def create(self, values: typing.List) -> Accumulator:
+        accumulators = [
+            accumulator_param.accumulator_type(
+                accumulator_param.constructor_params,
+                value) for accumulator_param, value in zip(
+                    self._accumulator_params, values)
```

Comment:
Accumulator constructor take `values` not `value` (and it's immediately starts to accumulate), so no need in zip(). 

The idea is that all accumulators will receive the same input values, i.e. in SQL terms they aggregate the same column. In case if the user needs to do aggregations on different columns, there will be multiple AccumulatorFactory objects

Reply:
Got it! Changed it.